routers/web/repo/fork.go

```diff
@@ -166,7 +166,11 @@ func ForkPost(ctx *context.Context) {
 			ctx.RenderWithErr(ctx.Tr("repo.settings.new_owner_has_same_repo"), tplFork, &form)
 			return
 		}
-		repo := repo_model.GetForkedRepo(ctx, ctxUser.ID, traverseParentRepo.ID)
+		repo, err := repo_model.GetForkedRepo(ctx, ctxUser.ID, traverseParentRepo.ID)
+		if err != nil && !repo_model.IsErrRepoNotExist(err) {
+			ctx.ServerError("GetForkedRepo", err)
+			return
+		}
```

Comment:
The "merge" made code more complex than it should be.

I really dislike (hate) so many "if err { ServerError; return }" in code.

Reply:
Do you have any better idea? Just skip the error when getting a fork like before.

Reply:
Leave to 1.24?